pkg/etcdcli/etcdcli.go

```diff
@@ -55,22 +58,26 @@ func (g *etcdClientGetter) getEtcdClient() (*clientv3.Client, error) {
 		etcdEndpoints = append(etcdEndpoints, fmt.Sprintf("https://%s:2379", internalIP))
 	}
 
-	hostEtcd, err := g.endpointsLister.Endpoints(operatorclient.TargetNamespace).Get("host-etcd")
+	hostEtcd, err := g.endpointsLister.Endpoints(operatorclient.TargetNamespace).Get("host-etcd-2")
 	if err != nil {
 		return nil, err
 	}
-	for _, addr := range hostEtcd.Subsets[0].Addresses {
-		if addr.Hostname == "etcd-bootstrap" {
-			// etcd-bootstrap has a valid IP in host-etcd
-			etcdEndpoints = append(etcdEndpoints, fmt.Sprintf("https://%s:2379", addr.IP))
-			break
+	bootstrapIP, ok := hostEtcd.Annotations[BootstrapIPAnnotationKey]
+	if !ok {
+		klog.V(2).Infof("service/host-etcd-2 is missing annotation %s", BootstrapIPAnnotationKey)
+	}
+	if bootstrapIP != "" {
+		// escape if IPv6
+		if net.ParseIP(bootstrapIP).To4() == nil {
+			bootstrapIP = "[" + bootstrapIP + "]"
```

Comment:
bad practice to have a variable named one thing, then mutated to mean something different.  After escaping, my understanding is that this is no longer an IP.